File: frontend/api/stats.ts
/**
 * 统计和站点配置相关 API
 */
import api from './config';

export interface StatsOverview {
    today_views: number;
    today_visitors: number;
    total_articles: number;
    total_comments: number;
    pending_comments: number;
    today_ai_calls: number;
    total_ai_calls: number;
}

export interface DailyStat {
    date: string;
    total_views: number;
    unique_visitors: number;
    article_views: number;
    new_comments: number;
    ai_api_calls: number;
}

export interface PopularArticle {
    id: number;
    title: string;
    view_count: number;
}

export interface SiteSetting {
    id: number;
    key: string;
    value?: string;
    type: string;
    description?: string;
}

export interface ContactInfo {
    email?: string;
    github?: string;
    bilibili?: string;
    qq?: string;
    wechat?: string;
    twitter?: string;
}

export interface PublicSettings {
    site_title?: string;
    site_description?: string;
    admin_avatar?: string;
    admin_bio?: string;
    contact: ContactInfo;
}

// 记录页面访问
export const recordPageView = async (data: {
    page_path: string;
    article_id?: number;
    referer?: string;
}): Promise<void> => {
    await api.post('/stats/page-view', data);
};

// 获取统计概览（管理员）
export const getStatsOverview = async (): Promise<StatsOverview> => {
    const response = await api.get('/stats/overview');
    return response.data;
};

// 获取每日统计（管理员）
export const getDailyStats = async (days?: number): Promise<DailyStat[]> => {
    const response = await api.get('/stats/daily', { params: { days } });
    return response.data;
};

// 获取热门文章（管理员）
export const getPopularArticles = async (limit?: number): Promise<PopularArticle[]> => {
    const response = await api.get('/stats/popular-articles', { params: { limit } });
    return response.data;
};

// 获取公开站点设置
export const getPublicSettings = async (): Promise<PublicSettings> => {
    const response = await api.get('/settings/public');
    return response.data;
};

// 获取所有设置（管理员）
export const getAllSettings = async (): Promise<SiteSetting[]> => {
    const response = await api.get('/settings');
    return response.data;
};

// 更新单个设置
export const updateSetting = async (key: string, value: string): Promise<SiteSetting> => {
    const response = await api.put(`/settings/${key}`, { value });
    return response.data;
};

// 批量更新设置
export const batchUpdateSettings = async (settings: Record<string, string>): Promise<SiteSetting[]> => {
    const response = await api.put('/settings/batch', { settings });
    return response.data;
};

File: backend/app/schemas/stats.py
"""
统计 Pydantic 模型
"""

from datetime import datetime, date
from typing import Optional, List

from pydantic import BaseModel, Field


class PageViewCreate(BaseModel):
    """记录页面访问请求模型"""
    page_path: str = Field(..., max_length=255, description="页面路径")
    article_id: Optional[int] = Field(None, description="文章 ID（如果是文章页）")
    referer: Optional[str] = Field(None, max_length=500, description="来源页面")


class DailyStatResponse(BaseModel):
    """每日统计响应模型"""
    date: date
    total_views: int = 0
    unique_visitors: int = 0
    article_views: int = 0
    new_comments: int = 0
    ai_api_calls: int = 0
    
    class Config:
        from_attributes = True


class DailyStatUpdate(BaseModel):
    """修改每日统计请求模型（增量式）"""
    date: str = Field(..., description="日期 YYYY-MM-DD")
    total_views: int = Field(0, description="访问次数增量")
    unique_visitors: int = Field(0, description="独立访客增量")
    article_views: int = Field(0, description="文章访问增量")
    new_comments: int = Field(0, description="新评论增量")
    ai_api_calls: int = Field(0, description="AI调用增量")


class StatsOverview(BaseModel):
    """统计概览响应模型"""
    today_views: int = 0
    today_visitors: int = 0
    total_articles: int = 0
    total_comments: int = 0
    pending_comments: int = 0
    today_ai_calls: int = 0
    total_ai_calls: int = 0


class PopularArticle(BaseModel):
    """热门文章模型"""
    id: int
    title: str
    view_count: int
    published_at: Optional[datetime] = None


class SiteSettingResponse(BaseModel):
    """站点配置响应模型"""
    key: str
    value: Optional[str] = None
    type: str
    description: Optional[str] = None
    
    class Config:
        from_attributes = True


class SiteSettingUpdate(BaseModel):
    """更新站点配置请求模型"""
    value: str = Field(..., description="配置值")


class ContactInfo(BaseModel):
    """联系方式信息"""
    email: Optional[str] = None
    wechat: Optional[str] = None
    github: Optional[str] = None
    qq: Optional[str] = None
    bilibili: Optional[str] = None


class PublicSiteSettings(BaseModel):
    """公开的站点配置"""
    site_title: str = "DevLog"
    site_description: str = ""
    admin_avatar: str = ""
    admin_bio: str = ""
    contact: ContactInfo = ContactInfo()
